### PR TITLE
add listtransactionsfrom RPC call

### PR DESCRIPTION
**Pull request by gandrewstone**
This behaves like listtransactions probably should have originally. The 'from' parameter starts at the oldest transaction.

### CODE REVIEW

**Comment by gandrewstone** [comment]
The only way to implement this is to iterate through all transactions in the wallet from the beginning
This is in theory inefficient, and may be why the original listtransactions code was written to uselessly count from the end.

But in practice the entire wallet history is loaded into RAM, so this API likely only gets inefficient at a scale when so many other things will go wrong.

If we rework the wallet to handle a history of 1x10^8+ transactions, we can still support this RPC by associating a count with each transaction so we'd no longer need to iterate through the whole thing.

**Review by ptschip — APPROVED** [review]
looks good.

File: qa/rpc-tests/listtransactions.py [hunk 1]
```diff
@@ -11,78 +11,151 @@
...
 
+
 def txFromHex(hexstring):
     tx = CTransaction()
     f = BytesIO(hex_str_to_bytes(hexstring))
     tx.deserialize(f)
     return tx
 
+
 class ListTransactionsTest(BitcoinTestFramework):
```
Comment (sickpig) [inline qa/rpc-tests/listtransactions.py:22]: nit: unrelated changes on L21 and L14

File: qa/rpc-tests/listtransactions.py [hunk 2]
```diff
@@ -11,78 +11,151 @@
...
+        # test when I advance to the end, I get nothing
+        curpos += len(tmp)
+        tmp = self.nodes[2].listtransactionsfrom("*", 100, curpos)
+        assert tmp == []
+
+    def test_listtransactions(self):
         # Simple send, 0 to 1:
         txid = self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 0.1)
         self.sync_all()
         assert_array_result(self.nodes[0].listtransactions(),
```
Comment (sickpig) [inline qa/rpc-tests/listtransactions.py:103]: nit: unrelated formatting changes fro m L31 to L169

**Review by sickpig — APPROVED** [review]
approved modulo unrelated changes formatting changes to listtransactions.py

File: qa/rpc-tests/listtransactions.py (hunk 2 above)
Comment (gandrewstone) [reply qa/rpc-tests/listtransactions.py:103]: I ran "autopep8 --max-line-length 120 --in-place <files>" on this.  We should generally be adhering to a python code formatting standard as well as a c++ one.

Comment (sickpig) [reply qa/rpc-tests/listtransactions.py:103]: we should and I think we should introduce an automatic check in travis to verify that the file are properly linted.

what I would do thou is using not mixing formating w/ functional changes, because it makes review more difficult for me. This is not the case of this PR cause is simple and straight forward but still.

Comment (gandrewstone) [reply qa/rpc-tests/listtransactions.py:103]: good point
